sdjournal/journal.go

```diff
@@ -938,3 +965,60 @@ func (j *Journal) GetUsage() (uint64, error) {
 
 	return uint64(out), nil
 }
+
+// GetUniqueValues returns all unique values for a given field.
+func (j *Journal) GetUniqueValues(field string) ([]string, error) {
+	var result []string
+
+	sd_journal_query_unique, err := getFunction("sd_journal_query_unique")
+	if err != nil {
+		return result, err
```

Comment:
Sorry to nit you, but error handling style again: it's preferable to explicitly return `nil, err`, rather than e.g. returning this implicitly-empty slice. Rather, move the result slice declaration down to line 1001-1002 just before it is used. (This also fixes the consistency of returning `result, err` up here but `nil, err` within the for loop below...)